Modules/+Experiments/+AutoExperiment/@SpecSlowScan/analyze.m

```diff
@@ -191,12 +196,37 @@
         'amplitudes',NaN,...
         'widths',NaN,...
         'locations',NaN,...
+        'etas',NaN,...
         'background',NaN,...
         'index',NaN,... % Index into sites
         'redo',false,...
         'ignore',[]);     % indices of experiments in the fit
 end
 
+if preanalyze
+    err = [];
+    try
+        site_index = 1;
+        progbar = waitbar(site_index/n,'','Name','Analyzing all data','CreateCancelBtn','setappdata(gcbf,''canceling'',1)');
+        setappdata(progbar,'canceling',0);
+        update_all()
+        for curr_index=2:n
+            waitbar(curr_index/n,progbar,sprintf('Analyzing site %i/%i',curr_index,n));
+            drawnow limitrate
+            changeSite(curr_index);
```

Comment:
suppose the user loaded an analysis file using the `'Analysis'` flag, should this skip sites that already have been analyzed? Perhaps it should error if the user does this (that might be kind of annoying if you wanted to "resume" later)?

Reply:
Hm, not sure. While that seems in principle a valid use-case, the pre-analysis doesn't take so long (I've run on ~300 sites and I just go pour myself some more coffee then it's done) that I'd imagine someone needing to stop it partway through then resume. In that of that, I'd be surprised if someone wanted to load an analysis file AND do pre-analysis.

Reply:
so what is the verdict? make it error?

Reply:
Actually I think the current behavior is ok? It'll go ahead and step through the site, but it won't change the analysis. Slightly slower, sure, but not dramatically, and the current flow seems like the "right" way to do it somehow (keeping things modular, all updates going through the standard methods).